pennylane/transforms/defer_measurements.py

```diff
@@ -127,29 +128,37 @@ def func(x, y):
         raise ValueError("Continuous variable operations and observables are not supported.")
 
     # Find wires that are reused after measurement
-    measured_wires = set()
+    measured_wires = []
     reused_measurement_wires = set()
+    repeated_measurement_wire = False
 
     for op in tape.operations:
         if isinstance(op, MidMeasureMP):
-            if op.wires[0] in measured_wires or op.reset is True:
+            if op.reset is True:
                 reused_measurement_wires.add(op.wires[0])
-            measured_wires.add(op.wires[0])
+
+            if op.wires[0] in measured_wires:
+                repeated_measurement_wire = True
+            measured_wires.append(op.wires[0])
 
         else:
             reused_measurement_wires = reused_measurement_wires.union(
-                measured_wires.intersection(op.wires.toset())
+                set(measured_wires).intersection(op.wires.toset())
             )
 
     # Apply controlled operations to store measurement outcomes and replace
     # classically controlled operations
     control_wires = {}
-    cur_wire = max(tape.wires) + 1 if reused_measurement_wires else None
+    cur_wire = (
+        max(tape.wires) + 1 if reused_measurement_wires or repeated_measurement_wire else None
+    )
 
-    for op in tape:
+    for op in tape.operations:
         if isinstance(op, MidMeasureMP):
-            # Only store measurement outcome in new wire if wire gets reused
-            if op.wires[0] in reused_measurement_wires:
+            _ = measured_wires.pop(0)
```

Comment:
what does popping here achieve? if the same wire is measured twice, do we want it to be popped the first time?